listenbrainz/mbid_mapping/mapping/spotify_mapping.py

```diff
@@ -0,0 +1,45 @@
+from datetime import datetime
+
+from sqlalchemy import text
+
+from listenbrainz.db import timescale
+
+base_query_begin = text("""
+    INSERT INTO spotify_mapping
+         SELECT data->'track_metadata'->'additional_info'->>'recording_msid'::UUID
+              , data->'track_metadata'->'additional_info'->>'spotify_id'
+              , data->'track_metadata'->'additional_info'->>'spotify_album_id'
+              , data->'track_metadata'->'additional_info'->>'spotify_artist_ids'
+              , data->'track_metadata'->'additional_info'->>'spotify_album_artist_ids'
+              , data->'track_metadata'->'additional_info'->>'isrc'
+           FROM listen
+          WHERE data->'track_metadata'->'additional_info'->>'recording_msid' IS NOT NULL
+            AND data->'track_metadata'->'additional_info'->>'spotify_id' IS NOT NULL
+            AND """)
+initial_query = text("listened_at >= :start AND listened_at < :end")
+incremental_query = text("created >= :watermark")
+# spotify_id field is considered while calculating a msid, so one msid can only have
+# one spotify track id associated with it. Assuming, given a spotify track id other
+# spotify ids do not change all ids for a given msid will be fixed. Thus, ignoring rows
+# if msid is already present in mapping is enough to avoid duplicates and not lose data.
+base_query_end = text(" ON CONFLICT (recording_msid) DO NOTHING")
+
+LISTEN_MINIMUM_TS = int(datetime(2002, 10, 1).timestamp())
```

Comment:
These constants are already exisiting in other open PRs, will update to use those if those get merged before this.